### DISCUSSION TITLE
Is the composite extension enabled in the release package?

### DISCUSSION THREAD

**Discussion by cfgillil** [XQuartz/XQuartz#255] [Q&A] [answered]
### If not, can it be enabled when building from source?

For some context, I've been playing around with using XQuartz with the BSPWM tiling window manager, which works great.
I've also managed to get dmenu to run, and picom to compile, but running picom fails - complaining that there is **"No composite extension"**.

I've been using the dmg release thus far, and I am wondering if I could enable the composite extension by building XQuartz from source?
Any feedback would be greatly appreciated!

**Comment by cfgillil** [comment]
After some more experimentation, I think I may have a better way to ask the question...

Using the dmg release of XQuartz 'out of the box', the X server provided by XQuartz.app does not support the Composite extension.
This is verified by running `xdpyinfo -d $DISPLAY -ext Composite`
which returns `Composite extension not supported by server`

Digging into the /opt/X11 directory, I came across /opt/X11/bin/Xorg, which seems to be an X server.
so I run Xorg and it fails with the following output
`(==) Using default built-in configuration (21 lines)
(EE)
Fatal server error:
(EE) no screens found(EE)`

So maybe something is wrong with this "default built-in configuration"
I then came across /opt/X11/etc/X11/dummy.conf, renamed it to xorg.conf, and tried again.

Success! The Xorg server started up, and the log even says that the Composite extension is loaded!
To verify this, I tried the xdpyinfo command again, and it confirms: `Composite version 0.4 opcode: 141`

I wanted to see this with my own eyes, so after painstakingly installing x11vnc with X11 support on my mac, indeed this Xorg supports Composite, and picom is happily doing its thing too. However, this is a _really_ hacky workaround, as XQuartz is _way_ more usable.

So my question is two-fold:
### Why does XQuartz ship with an Xorg that does support Composite, while its macOS application does not?
### Is there some ‘switch’ in the compilation process that would ‘turn on’ Composite in the macOS application?

**Comment by jeremyhu** [comment, marked as answer]
Xorg and Xquartz are two completely different DDXs.

Xorg does support COMPOSITE but requires a compositing window manager. There is no compositing window manager for macOS that is ready to ship. XtoQ and libxcwm was a great start, but it didn't get to a shippable state. We would need a bit of development support to bring that product to shipping in order to drop the Xquartz DDX and switch to Xorg.
